src/convertToObject.js

```diff
@@ -1,18 +1,36 @@
 'use strict';
 
 /**
- * Implement convertToObject function:
- *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
- * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
- *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const sourseArr = sourceString.split('\n');
+
+  const result = sourseArr.reduce((acc, el) => {
+    const parts = el.split(':');
+    const key = parts[0].trim();
+
+    let value = parts.slice(1).toString().trim();
+
+    if (value.endsWith(';')) {
+      value = value.slice(0, -1);
+    }
+
+    if (key === 'clear') {
+      value = value.trim();
+    }
+
+    acc[key] = value;
+
+    return acc;
+  }, {});
+
+  delete result[''];
+  delete result[';'];
```

Comment:
To avoid this, you can use method `.filter` for your array (Styles always containing ':' symbol)